src/Liquidity sawp Test.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {Test, console} from "forge-std/Test.sol";
+import {LiquidityPool} from "../src/liquidityPoolSwapFunction.sol";
+import {MockERC20} from "../src/MockERC20.sol"; // Import the custom MockERC20
+
+contract SwapFunctionTest is Test {
+    LiquidityPool liquidityPool;
+    MockERC20 token0;
+    MockERC20 token1;
+    MockERC20 lpRewardsToken; // Add this line
+    address initialOwner = address(0x789);
+    address user1 = address(0x123);
+    address user2 = address(0x456);
+
+    // Constants for test
+    string constant POOL_NAME = "TestPool";
+    uint24 constant FEE = 3000; // Example fee value
+    uint160 constant INITIAL_SQRT_PRICE_X96 = 79228162514264337593543950336; // sqrt(2^96)
+    uint128 constant INITIAL_LIQUIDITY = 1000;
+    uint256 constant LOW_PRICE = 100;
+    uint256 HIGH_PRICE = 200;
+    uint256 constant TOKEN_AMOUNT = 1000 * 10**18;
+    uint256 constant SWAP_AMOUNT_IN = 500 * 10**18;
+    uint256 constant SWAP_AMOUNT_OUT_MIN = 1;
+
+      // Constants for test
+    string constant POOL_NAME = "TestPool";
+    uint24 constant FEE = 3000; // Example fee value
+    uint256 constant LOW_PRICE = 100;
+    uint256 constant HIGH_PRICE = 200;
+    uint256 constant TOKEN_AMOUNT = 1000 * 10**18;
+
+    function setUp() external {
+        // Deploy mock tokens
+        token0 = new MockERC20("Token0", "TK0");
+        token1 = new MockERC20("Token1", "TK1");
+        lpRewardsToken = new MockERC20("LPReward", "LPR");
+
+        // Deploy the LiquidityPool contract
+        vm.prank(initialOwner);
+        liquidityPool = new LiquidityPool(
+            POOL_NAME,
+            address(token0),
+            address(token1),
+            FEE,
+            LOW_PRICE,
+            HIGH_PRICE
+        );
+
+        // Mint tokens to users
+        token0.mint(user1, TOKEN_AMOUNT);
+        token1.mint(user2, TOKEN_AMOUNT);
+
+        // Mint tokens to the liquidity pool
+        token0.mint(address(liquidityPool), TOKEN_AMOUNT);
+        token1.mint(address(liquidityPool), TOKEN_AMOUNT);
+
+        // Approve the pool to spend tokens on behalf of users
+        vm.prank(user1);
+        token0.approve(address(liquidityPool), type(uint256).max);
+
+        vm.prank(user2);
+        token1.approve(address(liquidityPool), type(uint256).max);
+    }
+
+    function testCreatePool() public {
+        // Verify the pool was created correctly
+        liquidityPool.createPool(POOL_NAME, INITIAL_SQRT_PRICE_X96, INITIAL_LIQUIDITY, LOW_PRICE, HIGH_PRICE);
+        LiquidityPool.Pool memory pool = liquidityPool.getPool(POOL_NAME);
+
+        assertEq(pool.sqrtPriceX96, INITIAL_SQRT_PRICE_X96);
+        assertEq(pool.liquidity, INITIAL_LIQUIDITY);
+        assertEq(pool.lowPrice, LOW_PRICE);
+        assertEq(pool.highPrice, HIGH_PRICE);
+        assertEq(pool.creator, initialOwner);
+        assertEq(pool.poolName, POOL_NAME);
+    }
+
+  
+
+
```

Comment:
Swap should call changeReserveThrough swap method that too fetching the required address of the pool from AllPoolManger.sol side

https://github.com/Web3-Ethiopia/DEX/blob/2671eb340a28195c9f0266c19d6fba1cb523819f/src/LiqidityPool.sol